src/Core/Checkout/Document/Zugferd/ZugferdDocument.php

```diff
@@ -0,0 +1,247 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Checkout\Document\Zugferd;
+
+use horstoeko\zugferd\codelists\ZugferdAllowanceCodes;
+use horstoeko\zugferd\codelists\ZugferdDutyTaxFeeCategories;
+use horstoeko\zugferd\codelists\ZugferdInvoiceType;
+use horstoeko\zugferd\codelists\ZugferdSchemeIdentifiers;
+use horstoeko\zugferd\codelists\ZugferdUnitCodes;
+use horstoeko\zugferd\ZugferdDocumentBuilder;
+use horstoeko\zugferd\ZugferdDocumentValidator;
+use Shopware\Core\Checkout\Cart\Tax\Struct\CalculatedTax;
+use Shopware\Core\Checkout\Cart\Tax\Struct\CalculatedTaxCollection;
+use Shopware\Core\Checkout\Document\DocumentConfiguration;
+use Shopware\Core\Checkout\Document\DocumentException;
+use Shopware\Core\Checkout\Order\Aggregate\OrderAddress\OrderAddressEntity;
+use Shopware\Core\Checkout\Order\Aggregate\OrderCustomer\OrderCustomerEntity;
+use Shopware\Core\Checkout\Order\Aggregate\OrderDelivery\OrderDeliveryCollection;
+use Shopware\Core\Checkout\Order\Aggregate\OrderLineItem\OrderLineItemEntity;
+use Shopware\Core\Checkout\Order\OrderEntity;
+use Shopware\Core\Checkout\Promotion\Aggregate\PromotionDiscount\PromotionDiscountEntity;
+use Shopware\Core\Framework\Log\Package;
+
+#[Package('checkout')]
+class ZugferdDocument
```

Comment:
I find this class to be a very weird mix between a DTO and actual functionality. Is there any reason, why you don't just put these methods as private methods of the ZugferdBuilder?